src/rgen/Microsoft.Macios.Generator/Emitters/EnumEmitter.cs

```diff
@@ -3,22 +3,156 @@
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Macios.Generator.Attributes;
 using Microsoft.Macios.Generator.Context;
+using Microsoft.Macios.Generator.Extensions;
+using ObjCBindings;
 
 namespace Microsoft.Macios.Generator.Emitters;
 
 #pragma warning disable CS9113 // Parameter is unread. This class is work in progress
-class EnumEmitter (ISymbolBindingContext<EnumDeclarationSyntax> context, TabbedStringBuilder builder)
+class EnumEmitter (SymbolBindingContext context, TabbedStringBuilder builder)
 #pragma warning restore CS9113 // Parameter is unread.
 	: ICodeEmitter {
 
+	public string SymbolNamespace => context.Namespace;
 	public string SymbolName => $"{context.SymbolName}Extensions";
+	public IEnumerable<string> UsingStatements => ["Foundation", "ObjCRuntime", "System"];
 
-	public IEnumerable<string> UsingStatements => [];
+	void Emit (TabbedStringBuilder classBlock, (IFieldSymbol Symbol, FieldData<EnumValue> FieldData) enumField, int index)
+	{
+		var typeNamespace = enumField.Symbol.ContainingType.ContainingNamespace.Name;
+		if (!context.RootBindingContext.TryComputeLibraryName (enumField.FieldData.LibraryName, typeNamespace,
+				out string? libraryName, out string? libraryPath)) {
+			return;
+		}
+
+		classBlock.AppendLine ($"[Field (\"{enumField.FieldData.SymbolName}\", \"{libraryPath ?? libraryName}\")]");
+
+		using (var propertyBlock = classBlock.CreateBlock ($"internal unsafe static IntPtr {enumField.FieldData.SymbolName}", true))
+		using (var getterBlock = propertyBlock.CreateBlock ("get", true)) {
+			getterBlock.AppendLine ($"fixed (IntPtr *storage = &values [{index}])");
+			getterBlock.AppendLine (
+				$"\treturn Dlfcn.CachePointer (Libraries.{libraryName}.Handle, \"{enumField.FieldData.SymbolName}\", storage);");
```

Comment:
I was thinking of adding that logic in the analyzer, since we can use it to throw errors on the build and the IDE. That way we kill two birds with one stone.

Reply:
LibraryName is indeed something we can check since it is provided by the user and can generate bad c# code, but SymbolName, although it can be wrong (spaces and what not), it won't generate bad code since it is between "". We can add rules about the symbol names in the roslyn analyzer. Does rules can, and probably will, be different that does of the c# identifiers since they are ObjC symbols, C or C++ symbols, but not C# ones.

I'll add a check for the library name.

Reply:
I added it for both, since the C# identifier rules are very close to those of the other langs.